utils/filesystem/files_posix.go

```diff
@@ -0,0 +1,21 @@
+//go:build unix || (js && wasm)
+// +build unix js,wasm
+
+package filesystem
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+)
+
+func determineFileOwners(info os.FileInfo) (uid, gid int, err error) {
+	if raw, ok := info.Sys().(*syscall.Stat_t); ok {
+		gid = int(raw.Gid)
+		uid = int(raw.Uid)
+	} else {
+		err = commonerrors.ErrNotImplemented
```

Comment:
same with above, in unix/linux at least stat will be implemented won't it? A failure would be another issue

Reply:
So filesystems ay not define Sys() as `Stat_t` even if running on Linux and in this case, we want to say that we don't know what to do. I case `ErrNotSupported` may be better. I'll change this

Reply:
I change the error returned with some better description